sdk/go/common/workspace/paths_test.go

```diff
@@ -142,3 +142,29 @@ func TestDetectProjectUnreadableParent(t *testing.T) {
 	_, _, err = DetectProjectAndPath()
 	assert.ErrorIs(t, err, ErrProjectNotFound)
 }
+
+//nolint:paralleltest // Theses test use and change the current working directory
+func TestDetectProjectStackDeploymentPath(t *testing.T) {
+	tmpDir := mkTempDir(t)
+	cwd, err := os.Getwd()
+	assert.NoError(t, err)
+	defer func() { err := os.Chdir(cwd); assert.NoError(t, err) }()
+	err = os.Chdir(tmpDir)
+	assert.NoError(t, err)
+
+	yamlPath := filepath.Join(tmpDir, "Pulumi.yaml")
+	yamlContents := "name: some_project\ndescription: Some project\nruntime: nodejs\n"
```

Comment:
Maybe it's just me but I feel like a multi-line string could be more readable here? So e.g.

```go
yamlContents := `name: some_project
description: Some project
runtime: nodejs`
```